client/cd-create-profile.c

```diff
@@ -681,6 +685,73 @@ cd_util_icc_set_metadata_coverage (CdIcc *icc, GError **error)
 	return TRUE;
 }
 
+static gboolean
+cd_util_adjust_creation_time (CdUtilPrivate *priv, const char *source_filename, GError **error)
+{
+	g_autoptr(GFile) source_file;
+	g_autoptr(GFileInfo) file_info;
+	g_autoptr(GDateTime) creation_time;
+	GTimeVal source_file_creation_time;
+	gint64 build_date = 0;
+	char *source_date_epoch_string;
+	unsigned long long source_date_epoch;
+	char *endptr;
+
+	/* adjust creation time to match the XML one or SOURCE_DATE_EPOCH if specified and older */
+	source_date_epoch_string = getenv ("SOURCE_DATE_EPOCH");
+	if (source_date_epoch_string != NULL) {
+		errno = 0;
+		source_date_epoch = strtoull (source_date_epoch_string, &endptr, 10);
+		if ((source_date_epoch == ERANGE && (source_date_epoch == ULLONG_MAX || source_date_epoch == 0))
+		|| (errno != 0 && source_date_epoch == 0)) {
+			g_set_error (error, 1, 0, "Environment variable $SOURCE_DATE_EPOCH: strtoull: %s\n", g_strerror (errno));
```

Comment:
This is waaay longer than 80 chars, wide. Also, GError messages never end with a newline.